### PR TITLE
KAFKA-16921: Migrate test of connect module to Junit5 (Runtime subpackage)

### PR DESCRIPTION
**Pull request by frankvicky**
Migrate test of connect module to Junit5

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/DistributedConfigTest.java [hunk 1]
```diff
@@ -43,18 +43,18 @@
...
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.mockito.Mockito.doReturn;
 import static org.mockito.Mockito.doThrow;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
-@RunWith(MockitoJUnitRunner.StrictStubs.class)
+@ExtendWith(MockitoExtension.class)
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/DistributedConfigTest.java:59]: Please add `@MockitoSettings(strictness = Strictness.STRICT_STUBS)` too

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/IncrementalCooperativeAssignorTest.java [hunk 2]
```diff
@@ -49,16 +49,16 @@
...
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.mockito.ArgumentMatchers.notNull;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
-@RunWith(MockitoJUnitRunner.StrictStubs.class)
+@ExtendWith(MockitoExtension.class)
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/IncrementalCooperativeAssignorTest.java:63]: please add `@MockitoSettings(strictness = Strictness.STRICT_STUBS)`

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/WorkerCoordinatorIncrementalTest.java [hunk 3]
```diff
@@ -116,19 +112,15 @@ public class WorkerCoordinatorIncrementalTest {
...
-    @Parameter
-    public ConnectProtocolCompatibility compatibility;
-
-    @Parameter(1)
-    public int expectedMetadataSize;
-
-    @Before
-    public void setup() {
+    public void init(ConnectProtocolCompatibility compatibility) {
+        System.err.println("init:" + compatibility);
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/WorkerCoordinatorIncrementalTest.java]: please remove this debug message

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/WorkerGroupMemberTest.java [hunk 4]
```diff
@@ -27,25 +27,25 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.mockito.Mockito.atLeastOnce;
 import static org.mockito.Mockito.doReturn;
 import static org.mockito.Mockito.spy;
 import static org.mockito.Mockito.verify;
 
-@RunWith(MockitoJUnitRunner.StrictStubs.class)
+@ExtendWith(MockitoExtension.class)
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/WorkerGroupMemberTest.java:50]: please add `@MockitoSettings(strictness = Strictness.STRICT_STUBS)`

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/errors/ErrorReporterTest.java [hunk 5]
```diff
@@ -55,19 +55,19 @@
...
+import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.junit.jupiter.api.Assertions.assertInstanceOf;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.Mockito.times;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.verifyNoMoreInteractions;
 import static org.mockito.Mockito.when;
 
-@RunWith(MockitoJUnitRunner.StrictStubs.class)
+@ExtendWith(MockitoExtension.class)
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/errors/ErrorReporterTest.java:72]: please add `@MockitoSettings(strictness = Strictness.STRICT_STUBS)`

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/errors/RetryWithToleranceOperatorTest.java [hunk 6]
```diff
@@ -80,7 +80,7 @@
 import static org.mockito.Mockito.verifyNoMoreInteractions;
 import static org.mockito.Mockito.when;
 
-@RunWith(MockitoJUnitRunner.StrictStubs.class)
+@ExtendWith(MockitoExtension.class)
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/errors/RetryWithToleranceOperatorTest.java:85]: please add `@MockitoSettings(strictness = Strictness.STRICT_STUBS)`

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/errors/WorkerErrantRecordReporterTest.java [hunk 7]
```diff
@@ -25,24 +25,24 @@
...
-import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
-@RunWith(MockitoJUnitRunner.StrictStubs.class)
+@ExtendWith(MockitoExtension.class)
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/errors/WorkerErrantRecordReporterTest.java:47]: please add `@MockitoSettings(strictness = Strictness.STRICT_STUBS)`

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/health/ConnectClusterStateImplTest.java [hunk 8]
```diff
@@ -33,13 +33,13 @@
...
-import static org.junit.Assert.assertNotSame;
-import static org.junit.Assert.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotSame;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.mockito.ArgumentMatchers.eq;
 import static org.mockito.Mockito.doAnswer;
 
-@RunWith(MockitoJUnitRunner.StrictStubs.class)
+@ExtendWith(MockitoExtension.class)
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/health/ConnectClusterStateImplTest.java:44]: please add `@MockitoSettings(strictness = Strictness.STRICT_STUBS)`

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/DelegatingClassLoaderTest.java [hunk 9]
```diff
@@ -17,23 +17,23 @@
...
 
-import static org.junit.Assert.assertSame;
-import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
-@RunWith(MockitoJUnitRunner.StrictStubs.class)
+@ExtendWith(MockitoExtension.class)
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/DelegatingClassLoaderTest.java:38]: please add `@MockitoSettings(strictness = Strictness.STRICT_STUBS)`

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/SynchronizationTest.java [hunk 10]
```diff
@@ -113,19 +111,14 @@ public synchronized void clear() {
         public synchronized void set(Predicate<T> predicate) {
             clear();
             this.predicate = predicate;
-            // As soon as the barrier is tripped, the barrier will be reset for the next round.
             barrier = new CyclicBarrier(2);
         }
 
-        /**
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/SynchronizationTest.java]: why removing those comments?

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/SynchronizationTest.java [hunk 11]
```diff
@@ -140,12 +133,6 @@ public void await(T obj) {
             }
         }
 
-        /**
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/SynchronizationTest.java]: why removing them?

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/ConnectRestServerTest.java [hunk 12]
```diff
@@ -56,15 +55,16 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.ArgumentMatchers.eq;
 import static org.mockito.Mockito.doReturn;
 
-@RunWith(MockitoJUnitRunner.StrictStubs.class)
+@ExtendWith(MockitoExtension.class)
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/ConnectRestServerTest.java:69]: please add `@MockitoSettings(strictness = Strictness.STRICT_STUBS)`

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/InternalRequestSignatureTest.java [hunk 13]
```diff
@@ -34,18 +34,18 @@
...
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.mockito.ArgumentMatchers.anyString;
 import static org.mockito.ArgumentMatchers.eq;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
-@RunWith(MockitoJUnitRunner.StrictStubs.class)
+@ExtendWith(MockitoExtension.class)
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/InternalRequestSignatureTest.java:50]: please add `@MockitoSettings(strictness = Strictness.STRICT_STUBS)`

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/RestClientTest.java [hunk 14]
```diff
@@ -105,26 +104,18 @@ private static <T> RestClient.HttpResponse<T> httpRequest(
         );
     }
 
-
-    @RunWith(Parameterized.class)
+    @ExtendWith(MockitoExtension.class)
     public static class RequestFailureParameterizedTest {
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/RestClientTest.java]: Could we flat those inner tests?

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorPluginsResourceTest.java [hunk 15]
```diff
@@ -97,7 +97,7 @@
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
-@RunWith(MockitoJUnitRunner.StrictStubs.class)
+@ExtendWith(MockitoExtension.class)
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorPluginsResourceTest.java:102]: please add `@MockitoSettings(strictness = Strictness.STRICT_STUBS)`

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorsResourceTest.java [hunk 16]
```diff
@@ -83,7 +82,7 @@
 import static org.mockito.Mockito.verifyNoMoreInteractions;
 import static org.mockito.Mockito.when;
 
-@RunWith(MockitoJUnitRunner.StrictStubs.class)
+@ExtendWith(MockitoExtension.class)
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorsResourceTest.java:87]: please add `@MockitoSettings(strictness = Strictness.STRICT_STUBS)`

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/resources/InternalConnectResourceTest.java [hunk 17]
```diff
@@ -41,16 +41,16 @@
...
+import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.mockito.Mockito.any;
 import static org.mockito.Mockito.doAnswer;
 import static org.mockito.Mockito.eq;
 import static org.mockito.Mockito.isNull;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
-@RunWith(MockitoJUnitRunner.StrictStubs.class)
+@ExtendWith(MockitoExtension.class)
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/resources/InternalConnectResourceTest.java:55]: please add `@MockitoSettings(strictness = Strictness.STRICT_STUBS)`

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/resources/LoggingResourceTest.java [hunk 18]
```diff
@@ -21,33 +21,33 @@
...
-import static org.junit.Assert.assertNull;
-import static org.junit.Assert.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
-@RunWith(MockitoJUnitRunner.StrictStubs.class)
+@ExtendWith(MockitoExtension.class)
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/resources/LoggingResourceTest.java:44]: please add `@MockitoSettings(strictness = Strictness.STRICT_STUBS)`

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/resources/RootResourceTest.java [hunk 19]
```diff
@@ -20,23 +20,23 @@
...
-import org.mockito.junit.MockitoJUnitRunner;
+import org.mockito.junit.jupiter.MockitoExtension;
 
-import static org.junit.Assert.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
-@RunWith(MockitoJUnitRunner.StrictStubs.class)
+@ExtendWith(MockitoExtension.class)
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/resources/RootResourceTest.java:35]: please add `@MockitoSettings(strictness = Strictness.STRICT_STUBS)`

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/standalone/StandaloneHerderTest.java [hunk 20]
```diff
@@ -110,7 +110,7 @@
 import static org.mockito.Mockito.when;
 import static org.mockito.Mockito.withSettings;
 
-@RunWith(MockitoJUnitRunner.StrictStubs.class)
+@ExtendWith(MockitoExtension.class)
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/standalone/StandaloneHerderTest.java:113]: please add `@MockitoSettings(strictness = Strictness.STRICT_STUBS)`

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/WorkerCoordinatorIncrementalTest.java (hunk 3 above)
Comment (m1a2st) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/WorkerCoordinatorIncrementalTest.java]: Do there need to print compatibility value?

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/standalone/StandaloneHerderTest.java [hunk 21]
```diff
@@ -141,24 +141,39 @@ private enum SourceSink {
     private final SampleConnectorClientConfigOverridePolicy
             noneConnectorClientConfigOverridePolicy = new SampleConnectorClientConfigOverridePolicy();
 
-    @Before
+    @BeforeEach
     public void setup() throws ExecutionException, InterruptedException {
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/standalone/StandaloneHerderTest.java]: Could you please adjust this method to accept a boolean to mock?
```java
    public void setup(boolean mockTransform) {
        herder = mock(StandaloneHerder.class, withSettings()
                .useConstructor(worker, WORKER_ID, KAFKA_CLUSTER_ID, statusBackingStore, new MemoryConfigBackingStore(transformer), noneConnectorClientConfigOverridePolicy, new MockTime())
                .defaultAnswer(CALLS_REAL_METHODS));
        createCallback = new FutureCallback<>();
        final ArgumentCaptor<Map<String, String>> configCapture = ArgumentCaptor.forClass(Map.class);
        if (mockTransform) when(transformer.transform(eq(CONNECTOR_NAME), configCapture.capture())).thenAnswer(invocation -> configCapture.getValue());
    }
```

**Review by chia7712 — COMMENTED** [review]
@frankvicky thanks for this PR

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/PluginScannerTest.java [hunk 22]
```diff
@@ -17,161 +17,156 @@
...
 
-    private PluginScanResult scan(Set<Path> pluginLocations) {
+    private PluginScanResult scan(ScannerType scannerType, Set<Path> pluginLocations) {
         ClassLoaderFactory factory = new ClassLoaderFactory();
         Set<PluginSource> pluginSources = PluginUtils.pluginSources(pluginLocations, PluginScannerTest.class.getClassLoader(), factory);
-        return scanner.discoverPlugins(pluginSources);
+        return this.scannerMap.get(scannerType).discoverPlugins(pluginSources);
     }
-
-}
```
Comment (m1a2st) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/PluginScannerTest.java:177]: Please revert this unrelated change

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/RestClientTest.java [hunk 23]
```diff
@@ -352,4 +373,4 @@ public int hashCode() {
             return Objects.hash(content);
         }
     }
-}
```
Comment (m1a2st) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/RestClientTest.java:355]: Please revert this unrelated change

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/SynchronizationTest.java (hunk 10 above)
Comment (frankvicky) [reply connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/SynchronizationTest.java]: by mistake 😢 
will roll it back.

**Comment by frankvicky** [comment]
Hi @chia7712 , @m1a2st 
I have do some changes, PTAL 😄

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/DistributedHerderTest.java [hunk 24]
```diff
@@ -147,7 +147,7 @@
 import static org.mockito.Mockito.when;
 import static org.mockito.Mockito.withSettings;
 
-@RunWith(MockitoJUnitRunner.StrictStubs.class)
+@ExtendWith(MockitoExtension .class)
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/DistributedHerderTest.java]: please add `@MockitoSettings(strictness = Strictness.STRICT_STUBS)`

Comment (chia7712) [reply connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/DistributedHerderTest.java]: also, please remove redundant "space"

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/standalone/StandaloneHerderTest.java [hunk 25]
```diff
@@ -110,7 +111,8 @@
 import static org.mockito.Mockito.when;
 import static org.mockito.Mockito.withSettings;
 
-@RunWith(MockitoJUnitRunner.StrictStubs.class)
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.STRICT_STUBS)
 @SuppressWarnings("unchecked")
 public class StandaloneHerderTest {
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/standalone/StandaloneHerderTest.java:116]: Could we reuse the existent code as much as possible? please take a look this example: 
[StandaloneHerderTest.txt](https://github.com/user-attachments/files/15855186/StandaloneHerderTest.txt)

**Review by chia7712 — COMMENTED** [review]
@frankvicky thanks for updated PR

**Comment by frankvicky** [comment]
Hi @chia7712, I have make some changes based on your example to keep `StandaloneHerderTest` has minimum changes during migration, PTAL 😸

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/PluginScannerTest.java [hunk 26]
```diff
@@ -17,161 +17,156 @@
...
 
-    @Parameterized.Parameters
-    public static Collection<Object[]> parameters() {
-        List<Object[]> values = new ArrayList<>();
-        for (ScannerType type : ScannerType.values()) {
-            values.add(new Object[]{type});
-        }
-        return values;
+    @BeforeEach
+    public void setScanner() {
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/PluginScannerTest.java]: Maybe we can parameterize scanner directly?
```java
    static Stream<PluginScanner> parameters() {
        return Stream.of(new ReflectionScanner(), new ServiceLoaderScanner());
    }
```

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/RestClientTest.java [hunk 27]
```diff
@@ -105,228 +106,235 @@ private static <T> RestClient.HttpResponse<T> httpRequest(
...
-            doReturn(req).when(httpClient).newRequest(anyString());
-            ConnectRestException e = assertThrows(ConnectRestException.class, () -> httpRequest(
-                    httpClient, MOCK_URL, TEST_METHOD, TEST_TYPE, TEST_SIGNATURE_ALGORITHM
-            ));
-            assertIsInternalServerError(e);
-            assertInstanceOf(InterruptedException.class, e.getCause());
-            assertTrue(Thread.interrupted());
+    private String toJsonString(Object obj) {
+        try {
+            return OBJECT_MAPPER.writeValueAsString(obj);
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/RestClientTest.java]: we can wrap it by `assertDoesNotThrow`

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/WorkerGroupMemberTest.java [hunk 28]
```diff
@@ -27,25 +27,28 @@
 import org.apache.kafka.connect.runtime.WorkerConfig;
 import org.apache.kafka.connect.storage.ConfigBackingStore;
 import org.junit.Test;
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/WorkerGroupMemberTest.java]: please update this package

**Review by chia7712 — COMMENTED** [review]
@frankvicky thanks for updated PR

**Comment by frankvicky** [comment]
Hi @chia7712, I have make some changes based on feedback, PTAL 🔹

**Review by chia7712 — APPROVED** [review]
LGTM
